GDJS/Runtime/variable.js

```diff
@@ -4,21 +4,36 @@
  * This project is released under the MIT License.
  */
 
+/**
+ * @typedef variableData Data representation of a GDevelop variable
+ * @property {string} [name] The name of the variable. Used if a child variable.
+ * @property {string | number} [value] The value of the variable, either string or number. Leave blank for structures.
+ * @property {variableData} [children] The children of the structure. Leave blank if value is defined.
+ */
+
 /**
  * A Variable is an object storing a value (number or a string) or children variables.
  *
  * @memberof gdjs
  * @class Variable
- * @param {Object} varData optional object used to initialize the variable.
+ * @param {Object} varData Optional object used to initialize the variable.
+ * @param {Array<variableData>} varData.value A representation of a variable.
  */
 gdjs.Variable = function(varData)
 {
-    this._value = 0;
-    this._str = "";
-    this._numberDirty = false;
-    this._stringDirty = true;
+	/** @type {number} */
```

Comment:
These types are inferred automatically from the fact that you're setting the value - so strictly speaking this is not necessary :)

Reply:
That is only for the editor if you wan't the js code event to tell you what type is expected if you want to play around with it even if it is internal